redbeams/src/main/java/com/sequenceiq/redbeams/converter/stack/AllocateDatabaseServerV4RequestToDBStackConverter.java

```diff
@@ -288,18 +281,14 @@ private Json getTags(DBStack dbStack, AllocateDatabaseServerV4Request dbRequest,
 
         Map<String, String> defaultTags = costTagging.prepareDefaultTags(request);
         Map<String, String> environmentUserTags = Objects.requireNonNullElse(environment.getTags().getUserDefined(), new HashMap<>());
-        defaultTags.forEach((key, value) -> {
-            resultTags.put(key, value);
-        });
-        environmentUserTags.forEach((key, value) -> {
-            resultTags.putIfAbsent(key, value);
-        });
+        resultTags.putAll(defaultTags);
+        environmentUserTags.forEach(resultTags::putIfAbsent);
 
         return new Json(new StackTags(resultTags, new HashMap<>(), defaultTags));
     }
 
     private Map<String, String> getTags(Map<String, String> tags) {
-        return Objects.requireNonNullElse(tags, new HashMap<>());
+        return Objects.requireNonNullElse(new HashMap<>(tags), new HashMap<>());
```

Comment:
`Objects.requireNonNullElse` this doesn't make any sense, as it never would be `null`

Reply:
I can happily remove it, but `tags` is coming from `AllocateDatabaseServerV4Request.getTags()`. As you can see, `requireNonNullElse()` was already in place, I just added the object copy logic.

Reply:
yes, and thats what makes it unnecessary: 
```
Returns the first argument if it is non-null and otherwise returns the non-null second argument.
```
it would never be null, as `new HashMap<>(tags)` would never be null

Reply:
Ah, I see what you mean. :smile: NPE would result (in the `HashMap` constructor) if `tags` happened to be `null` for any reason, so I will replace `requireNonNullElse()` with a conditional construct.